File: fix-config.js
const fs = require('fs');
const path = require('path');

// Read the current config file
const configPath = path.join(__dirname, 'src/data/config.ts');
let content = fs.readFileSync(configPath, 'utf8');

console.log('Original file size:', content.length);

// Simple approach: use JSON parsing and reconstruction
// First, extract the export statement and parse
const exportMatch = content.match(/export const config = ({[\s\S]*});/);
if (!exportMatch) {
  console.error('Could not find export statement');
  process.exit(1);
}

const jsonStr = exportMatch[1];
try {
  const config = JSON.parse(jsonStr);
  
  // Recursive function to convert blocks to text
  function convertBlocks(obj) {
    if (obj === null || obj === undefined) return obj;
    if (typeof obj === 'string') return obj;
    
    if (Array.isArray(obj)) {
      // Check if this is a block array
      if (obj.length > 0 && obj[0]._type === 'block') {
        // Convert blocks to plain text
        return obj.map(block => {
          if (block.children && block.children.length > 0) {
            return block.children.map(child => child.text || '').join('');
          }
          return '';
        }).join('\n\n');
      }
      return obj.map(convertBlocks);
    }
    
    if (typeof obj === 'object') {
      const result = {};
      for (const [key, value] of Object.entries(obj)) {
        if (key === 'id' && value && typeof value === 'object' && value._type === 'slug') {
          // Convert slug objects to strings
          result[key] = value.current;
        } else {
          result[key] = convertBlocks(value);
        }
      }
      return result;
    }
    
    return obj;
  }
  
  const converted = convertBlocks(config);
  
  // Reconstruct with TypeScript export
  const output = 'export const config = ' + JSON.stringify(converted, null, 2) + ';';
  
  fs.writeFileSync(configPath, output, 'utf8');
  console.log('✅ Config file converted successfully');
  console.log('New file size:', output.length);
} catch (error) {
  console.error('Error:', error.message);
  process.exit(1);
}

File: fix_config.py
import json
import re
from pathlib import Path

config_path = Path('src/data/config.ts')
content = config_path.read_text(encoding='utf-8')

# Extract JSON part
match = re.search(r'export const config = ({.*});', content, re.DOTALL)
if not match:
    print("Could not find export statement")
    exit(1)

json_str = match.group(1)

# Parse JSON
try:
    config = json.loads(json_str)
except json.JSONDecodeError as e:
    print(f"JSON parsing error: {e}")
    exit(1)

# Recursive function to convert blocks to plain text
def convert_blocks(obj):
    if obj is None:
        return obj
    
    if isinstance(obj, str):
        return obj
    
    if isinstance(obj, list):
        # Check if this is a block array (each element has _type: 'block')
        if obj and isinstance(obj[0], dict) and obj[0].get('_type') == 'block':
            # Convert block array to plain text
            texts = []
            for block in obj:
                if 'children' in block and isinstance(block['children'], list):
                    block_text = ''.join(
                        child.get('text', '') 
                        for child in block['children']
                        if isinstance(child, dict)
                    )
                    if block_text:
                        texts.append(block_text)
            return '\n\n'.join(texts) if texts else None
        return [convert_blocks(item) for item in obj]
    
    if isinstance(obj, dict):
        result = {}
        for key, value in obj.items():
            if key == 'id' and isinstance(value, dict) and value.get('_type') == 'slug':
                # Convert slug object to string
                result[key] = value.get('current', '')
            else:
                result[key] = convert_blocks(value)
        return result
    
    return obj

converted = convert_blocks(config)

# Write back as TypeScript
output = 'export const config = ' + json.dumps(converted, indent=2) + ';'
config_path.write_text(output, encoding='utf-8')

print(f'✅ Config file converted successfully')
print(f'Original file size: {len(content)} bytes')
print(f'New file size: {len(output)} bytes')
